unified-runtime/source/adapters/opencl/queue.hpp

```diff
@@ -0,0 +1,61 @@
+//===--------- queue.hpp - OpenCL Adapter ---------------------------===//
+//
+// Copyright (C) 2025 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "common.hpp"
+#include "context.hpp"
+#include "device.hpp"
+
+#include <vector>
+
+struct ur_queue_handle_t_ {
+  using native_type = cl_command_queue;
+  native_type CLQueue;
+  ur_context_handle_t Context;
+  ur_device_handle_t Device;
+  // Used to keep a handle to the default queue alive if it is different
+  std::optional<ur_queue_handle_t> DeviceDefault = std::nullopt;
+  std::atomic<uint32_t> RefCount = 0;
+  bool IsNativeHandleOwned = true;
+
+  ur_queue_handle_t_(native_type Queue, ur_context_handle_t Ctx,
+                     ur_device_handle_t Dev)
+      : CLQueue(Queue), Context(Ctx), Device(Dev) {
+    RefCount = 1;
+    if (Device) {
```

Comment:
this is also kind of like the event-queue thing, but might be a bit trickier with ownership of the new handle due to the fact we could be dealing with a subdevice